commercetools-internal-processors/src/main/java/io/sphere/sdk/annotations/processors/generators/UpdateActionsGenerator.java

```diff
@@ -0,0 +1,107 @@
+    protected String getPackageName(final TypeElement annotatedTypeElement) {
+        return super.getPackageName(annotatedTypeElement).concat(".commands.updateactions");
+    }
+
+    @Override
+    public List<TypeSpec> generateTypes(final TypeElement annotatedTypeElement) {
+        final List<ExecutableElement> propertyMethods = getPropertyMethodsSorted(annotatedTypeElement);
+        final List<TypeSpec> typeSpecList = propertyMethods.stream()
+                .filter(m -> isPrimitiveType(PropertyGenModel.of(m).getType()))
+                .map(propertyMethod -> {
+                    final PropertyGenModel property = PropertyGenModel.of(propertyMethod);
+                    final MethodSpec getMethod = createGetMethodBuilder(propertyMethod).build();
+                    final String actionPrefix = property.isOptional() ? "set" : "change";
+                    final FieldSpec fieldSpec = createFieldBuilder(property, Modifier.PRIVATE)
+                            .addModifiers(Modifier.FINAL)
+                            .build();
+                    final String updateAction = actionPrefix + StringUtils.capitalize(fieldSpec.name);
+                    final String updateActionClassName = StringUtils.capitalize(updateAction);
+                    final TypeSpec.Builder typeSpecBuilder = TypeSpec.classBuilder(updateActionClassName)
+                            .addJavadoc("$L $L to $L\n", property.isOptional() ? "Sets" : "Updates", fieldSpec.name, ClassName.get(annotatedTypeElement).simpleName())
+                            .addJavadoc("\n")
+                            .addJavadoc("{@doc.gen intro}\n")
+                            .superclass(ParameterizedTypeName.get(ClassName.get(UpdateActionImpl.class), ClassName.get(annotatedTypeElement)))
+                            .addModifiers(Modifier.PUBLIC, Modifier.FINAL)
+                            .addAnnotation(AnnotationSpec.builder(Generated.class)
+                                    .addMember("value", "$S", getClass().getCanonicalName())
+                                    .addMember("comments", "$S", "Generated from: " + annotatedTypeElement.getQualifiedName().toString()).build())
+                            .addField(createFieldBuilder(property, Modifier.PRIVATE)
+                                    .addModifiers(Modifier.FINAL)
+                                    .build())
+                            .addMethod(createConstructor(property, updateAction))
+                            .addMethod(getMethod)
+                            .addMethod(createFactoryMethod(property, updateActionClassName));
+                    if (property.isOptional()) {
+                        typeSpecBuilder.addMethod(createUnsetMethod(updateActionClassName));
+                    }
+                    return typeSpecBuilder.build();
+                }).collect(Collectors.toList());
+
+        return typeSpecList;
+    }
+
+    protected boolean isPrimitiveType(final TypeName propertyType) {
```

Comment:
Cood idea to extract this into a method 👍 If we in the future need this method in other classes, we can still move it to our `TypeUtils` class.